pyproject.toml

```diff
@@ -8,34 +8,259 @@ path = "src/MaxText/__init__.py"
 [project]
 name = "MaxText"
 description = "MaxText is a simple, performant and scalable Jax LLM!"
-dynamic = ["dependencies", "optional-dependencies", "version"]
-requires-python = ">=3.10"
+dynamic = ["version"]
+requires-python = ">=3.12"
 readme = "README.md"
 license = "Apache-2.0"
 keywords = ["llm", "jax", "llama", "mistral", "mixtral", "gemma", "deepseek"]
 classifiers = [
     "Development Status :: 4 - Beta",
     "Programming Language :: Python",
-    "Programming Language :: Python :: 3.10",
-    "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
     "Programming Language :: ML",
     "Environment :: GPU"
 ]
+dependencies = [
+    "absl-py>=2.3.1",
+    "aiofiles>=24.1.0",
+    "aiohappyeyeballs>=2.6.1",
+    "aiohttp>=3.12.15",
+    "aiosignal>=1.4.0",
+    "annotated-types>=0.7.0",
```

Comment:
How do you decide the right JAX commit for the command you shared in the PR description? Is it possible to tie this to a JAX version instead?

```
seed-env --local-requirements=requirements.txt --seed-commit=5a4a3b9147811616f4d8ae2c69240d4685d6673f --python-version=3.12
```